src/main/java/net/sf/jabref/gui/shared/OpenSharedDatabaseDialog.java

```diff
@@ -300,12 +322,24 @@ private void initLayout() {
     /**
      * Saves the data from this dialog persistently to facilitate the usage.
      */
-    public void setGlobalPrefs() {
-        Globals.prefs.put(SHARED_DATABASE_TYPE, ((DBMSType) dbmsTypeDropDown.getSelectedItem()).toString());
-        Globals.prefs.put(SHARED_DATABASE_HOST, hostField.getText());
-        Globals.prefs.put(SHARED_DATABASE_PORT, portField.getText());
-        Globals.prefs.put(SHARED_DATABASE_NAME, databaseField.getText());
-        Globals.prefs.put(SHARED_DATABASE_USER, userField.getText());
+    public void setPreferences() {
+        prefs.setType(((DBMSType) dbmsTypeDropDown.getSelectedItem()).toString());
+        prefs.setHost(hostField.getText());
+        prefs.setPort(portField.getText());
+        prefs.setName(databaseField.getText());
+        prefs.setUser(userField.getText());
+
+        if (rememberPassword.isSelected()) {
+            try {
+                prefs.setPassword(new Password(new String(passwordField.getPassword()), userField.getText()).encrypt());
```

Comment:
With this line your are probably (breaking/reducing the security):
And a char array "protects" you from accidentally writing the password in plaintext in a log...
http://stackoverflow.com/questions/8881291/why-is-char-preferred-over-string-for-passwords-in-java

sso I would recommend using a char[] in your Password class